app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java

```diff
@@ -218,8 +219,12 @@ private void setupRakamFirstLaunchSuperProperty(boolean isFirstLaunch) {
   private void setupDimensions(android.app.Application application) {
     if (!checkForUTMFileInMetaINF(application)) {
       setUTMDimensionsToUnknown();
+      sendRakamUserProperties(UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN,
+          application.getPackageName());
     } else {
       setUserProperties(utmSource, utmMedium, utmCampaign, utmContent, entryPoint);
+      sendRakamUserProperties(utmContent, utmSource, utmCampaign, utmMedium, entryPoint,
+          application.getPackageName());
```

Comment:
why don't we set them in the setUserProperties anyway? shouldn't the setUserProperties deal with all the user properties from different analytics?

Reply:
I didnt want to merge both methods to keep rakam events sending independent

Reply:
They can be independent, but UserProperties is a parent of RakamUserProperties, all names should be at the same level